src/util/system.cpp

```diff
@@ -686,14 +686,12 @@ fs::path StripRedundantLastElementsOfPath(const fs::path& path)
 }
 } // namespace
 
-static fs::path g_blocks_path_cache_net_specific;
-static fs::path pathCached;
-static fs::path pathCachedNetSpecific;
-static RecursiveMutex csPathCached;
+static Mutex g_blocksdir_path_mutex;
+static fs::path g_blocks_path_cache_net_specific GUARDED_BY(g_blocksdir_path_mutex);
 
 const fs::path &GetBlocksDir()
 {
-    LOCK(csPathCached);
+    LOCK(g_blocksdir_path_mutex);
     fs::path &path = g_blocks_path_cache_net_specific;
```

Comment:
Assuming `g_blocksdir_path_mutex` protects just the variable `g_blocks_path_cache_net_specific`, would it be possible to release the mutex after this assignment?

If "yes", then I would suggest to do it as a good practice of minimizing the time a mutex is held and also to avoid holding it together with the other newly introduced `g_datadir_path_mutex`.

If "no" then I think it is warranted to explain in a comment next to `g_blocksdir_path_mutex` what exactly is protected by it.

Reply:
> Assuming `g_blocksdir_path_mutex` protects just the variable `g_blocks_path_cache_net_specific`, would it be possible to release the mutex after this assignment?

I guess not, as the `path` references to the same memory location as the `g_blocks_path_cache_net_specific` variable, that must be protected by the `g_blocksdir_path_mutex`.

Reply:
> If "no" then I think it is warranted to explain in a comment next to `g_blocksdir_path_mutex` what exactly is protected by it.

I think `GUARDED_BY` is pretty self-documented annotation, no?

Reply:
oh well, I overlooked the fact that it is being assigned to a reference